test/system/users_test.rb

```diff
@@ -0,0 +1,70 @@
+require 'application_system_test_case'
+
+class UsersTest < ApplicationSystemTestCase
+  setup do
+    @emails = ActionMailer::Base.deliveries
+    @emails.clear
+    @user = create :user
+    login_as @user
+    visit new_user_path
+  end
+
+  test 'invalid superadmin information' do
+    fill_in 'Email', with: ''
+    select '', from: 'Role'
+    click_button 'Create User'
```

Comment:
Try this:
```ruby
assert_no_difference 'User.count' do
  click_button 'Create User'
end
```
maybe you need to add `assert page.has_text? 'Please review the problems below:'` to delay the call to `assert_no_difference`

Then you can remove the `assert_equal User.last, @user` call